src/Microsoft.Extensions.Logging.Console/ConsoleLoggerProvider.cs

```diff
@@ -45,20 +45,32 @@ public ConsoleLoggerProvider(IConsoleLoggerSettings settings)
 
         private void OnConfigurationReload(object state)
         {
-            // The settings object needs to change here, because the old one is probably holding on
-            // to an old change token.
-            _settings = _settings.Reload();
+            try
+            {
+                // The settings object needs to change here, because the old one is probably holding on
+                // to an old change token.
+                _settings = _settings.Reload();
 
-            foreach (var logger in _loggers.Values)
+                foreach (var logger in _loggers.Values)
+                {
+                    logger.Filter = GetFilter(logger.Name, _settings);
```

Comment:
I am OK with that. My suggestion was "ideal world" adding more caching of data doesn't seem worth it.

IIRC the worst case here is that you end up with the first provider in a bad state, and probably broken. But for things like miss-typing an enum you wouldn't actually get any corruption of settings. Right?